apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.UnitTests/TypeTests.cs

```diff
@@ -0,0 +1,627 @@
+            yield return false;
+            yield return true;
+        }
+
+        private static IEnumerable<DateTime> GetDatesForArray()
+        {
+            yield return new DateTime(2017, 1, 31);
+            yield return new DateTime(2016, 2, 15);
+            yield return new DateTime(2015, 3, 31);
+        }
+
+        private static IEnumerable<DateTime> GetTimestampsForArray()
+        {
+            yield return new DateTime(2017, 1, 31, 3, 15, 30, 500);
+            yield return new DateTime(2016, 2, 15, 13, 15, 30, 000);
+            yield return new DateTime(2015, 3, 31, 3, 15, 30, 250);
+        }
+
+        // This data serves as inputs to converting from CLR types that a developer
+        // would use in his app to a given json serialized format using the spanner type
+        // to determine how to serialize.
+        // The inputs are used to test both ways (also deseralizing the generated json
+        // to a requested clr type).  However some cases are specified as only one direction
+        // usually because the conversion is by definition lossy.
+        public static IEnumerable<object[]> GetValidValueConversions()
+        {
+            //format is:  LocalClrInstance,  SpannerType,  SerializedJsonFromProto, [test one or both ways]
+            //testing can be one way if there is loss of information in the conversion.
+
+            //Spanner type = Float64 tests.
+            yield return new object[] {true, SpannerDbType.Float64, "1"};
+            yield return new object[] {false, SpannerDbType.Float64, "0"};
+            yield return new object[] {(byte) 1, SpannerDbType.Float64, "1"};
+            yield return new object[] {(sbyte) 1, SpannerDbType.Float64, "1"};
+            yield return new object[] {1.5M, SpannerDbType.Float64, "1.5"};
+            yield return new object[] {1.5D, SpannerDbType.Float64, "1.5"};
+            yield return new object[] {1.5F, SpannerDbType.Float64, "1.5"};
+            yield return new object[] {double.NegativeInfinity, SpannerDbType.Float64, S("-Infinity")};
+            yield return new object[] {double.PositiveInfinity, SpannerDbType.Float64, S("Infinity")};
+            yield return new object[] {double.NaN, SpannerDbType.Float64, S("NaN")};
+            yield return new object[] {1, SpannerDbType.Float64, "1"};
+            yield return new object[] {1U, SpannerDbType.Float64, "1"};
+            yield return new object[] {1L, SpannerDbType.Float64, "1"};
+            yield return new object[] {(ulong) 1, SpannerDbType.Float64, "1"};
+            yield return new object[] {(short) 1, SpannerDbType.Float64, "1"};
+            yield return new object[] {(ushort) 1, SpannerDbType.Float64, "1"};
+            yield return new object[] {"1", SpannerDbType.Float64, "1"};
+            yield return new object[] {"1.5", SpannerDbType.Float64, "1.5"};
+            yield return new object[] {null, SpannerDbType.Float64, "null"};
+
+            //Spanner type = Int64 tests.
```

Comment:
true -- originally some were lossy in precision and I'll add some back and make them one way.